hackathon_site/dashboard/frontend/src/slices/event/teamAdminSlice.test.ts

```diff
@@ -0,0 +1,66 @@
+import store, { makeStore, RootState } from "../store";
+import {
+    teamAdminReducerName,
+    initialState,
+    teamAdminSliceSelector,
+    isLoadingSelector,
+    getAllTeams,
+    teamAdminSelectors,
+} from "./teamAdminSlice";
+import { makeMockApiListResponse, waitFor } from "../../testing/utils";
+import { mockTeams } from "../../testing/mockData";
+import { get } from "../../api/api";
+
+jest.mock("api/api", () => ({
+    ...jest.requireActual("api/api"),
+    get: jest.fn(),
+}));
+const mockedGet = get as jest.MockedFunction<typeof get>;
+
+const mockState: RootState = {
+    ...store.getState(),
+    [teamAdminReducerName]: initialState,
+};
+
+describe("Selectors", () => {
+    test("teamAdminSliceSelector returns the isLoading state", () => {
+        expect(teamAdminSliceSelector(mockState)).toEqual(
+            mockState[teamAdminReducerName]
+        );
+    });
+
+    test("isLoadingSelector", () => {
+        const loadingTrueState = {
+            ...mockState,
+            [teamAdminReducerName]: {
+                ...initialState,
+                isLoading: true,
+            },
+        };
+        const loadingFalseState = {
+            ...mockState,
+            [teamAdminReducerName]: {
+                ...initialState,
+                isLoading: false,
+            },
+        };
+
+        expect(isLoadingSelector(loadingTrueState)).toEqual(true);
+        expect(isLoadingSelector(loadingFalseState)).toEqual(false);
+    });
+});
+
+describe("getAllTeams thunk", () => {
+    it("Updates the store on API success", async () => {
```

Comment:
you should add a test that checks that a snackbar has dispatched and the error state is updated correctly

Reply:
done